File: 2015/gara3/festa/att/festa.pas
function invita(N, M: longint; var conoscenzeA, conoscenzeB: array of longint): longint;
begin
    (* Mettete qui il codice della soluzione *)
    invita := 42;
end;


const
    MAXM = 100000;
var
    N, M, i: longint;
    conoscenzeA, conoscenzeB : array[0..MAXM-1] of longint;
    fr, fw : text;

begin
    assign(fr, 'input.txt');
    assign(fw, 'output.txt');
    reset(fr);
    rewrite(fw);
    readln(fr, N, M);
    for i:=0 to M-1 do
        readln(fr, conoscenzeA[i], conoscenzeB[i]);

    writeln(fw, invita(N, M, conoscenzeA, conoscenzeB));
    close(fr);
    close(fw);
end.

File: ed1/gara3/festa/att/festa.c
#include <stdio.h>
#include <assert.h>

#define MAXM 100000

int invita(int N, int M, int conoscenzeA[], int conoscenzeB[]) {
    // Mettete qui il codice della soluzione
    return 42;
}


int conoscenzeA[MAXM], conoscenzeB[MAXM];

int main() {
    FILE *fr, *fw;
    int N, M, i;

    fr = fopen("input.txt", "r");
    fw = fopen("output.txt", "w");
    assert(2 == fscanf(fr, "%d%d", &N, &M));
    for(i=0; i<M; i++)
        assert(2 == fscanf(fr, "%d%d", &conoscenzeA[i], &conoscenzeB[i]));

    fprintf(fw, "%d\n", invita(N, M, conoscenzeA, conoscenzeB));
    fclose(fr);
    fclose(fw);
    return 0;
}
